modules/image-loading-optimization/module.php

```diff
@@ -0,0 +1,346 @@
+	 * This function behaves more like an action than a filter.
+	 *
+	 * @param string $content the content.
+	 * @return string We simply return the content from parameter.
+	 */
+	public function flush_header_buffer( $content ) {
+		$buffer_status = ob_get_status();
+
+		if ( ! empty( $buffer_status ) &&
+			1 === $buffer_status['type'] &&
+			get_class( $this ) . '::handle_buffer_content' === $buffer_status['name'] ) {
+			ob_end_flush();
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Callback to handle image optimization logic on buffered content.
+	 *
+	 * @param string $buffer Buffered content.
+	 * @return string Content with optimized images.
+	 */
+	public function handle_buffer_content( $buffer ) {
+		return $this->filter_images( $buffer );
+	}
+
+	/**
+	 * Check for image in the content provided and apply optimization logic on them.
+	 *
+	 * @param string $content Content to be analyzed.
+	 * @return string Content with optimized images.
+	 */
+	private function filter_images( $content ) {
+		return preg_replace_callback(
+			'/<img\s[^>]+>/',
+			function ( $matches ) {
+				return $this->loading_optimization_image( $matches[0] );
+			},
+			$content
+		);
+	}
+
+	/**
+	 * Apply loading optimization logic on the image.
+	 *
+	 * @param mixed $image Original image tag.
+	 * @return string Optimized image.
+	 */
+	public function loading_optimization_image( $image ) {
+		if ( isset( self::$image_visited[ $image ] ) ) {
+			return self::$image_visited[ $image ];
+		}
+
+		$optimized_image = $this->add_loading_optimization_attrs( $image );
+		self::$image_visited[ $image ] = $optimized_image;
+
+		return $optimized_image;
+	}
+
+	/**
+	 * Adds optimization attributes to an `img` HTML tag.
+	 *
+	 * @param string $image   The HTML `img` tag where the attribute should be added.
+	 * @return string Converted `img` tag with optimization attributes added.
+	 */
+	private function add_loading_optimization_attrs( $image ) {
+		$width             = preg_match( '/ width=["\']([0-9]+)["\']/', $image, $match_width ) ? (int) $match_width[1] : null;
+		$height            = preg_match( '/ height=["\']([0-9]+)["\']/', $image, $match_height ) ? (int) $match_height[1] : null;
+		$loading_val       = preg_match( '/ loading=["\']([A-Za-z]+)["\']/', $image, $match_loading ) ? $match_loading[1] : null;
+		$fetchpriority_val = preg_match( '/ fetchpriority=["\']([A-Za-z]+)["\']/', $image, $match_fetchpriority ) ? $match_fetchpriority[1] : null;
+
+		// Images should have height and dimension width for the loading optimization attributes to be added.
+		if ( ! str_contains( $image, ' width="' ) || ! str_contains( $image, ' height="' ) ) {
+			return $image;
+		}
+
+		$optimization_attrs = $this->get_loading_optimization_attributes(
+			array(
+				'width'         => $width,
+				'height'        => $height,
+				'loading'       => $loading_val,
+				'fetchpriority' => $fetchpriority_val,
+			)
```

Comment:
```suggestion
			[
				'width'         => $width,
				'height'        => $height,
				'loading'       => $loading_val,
				'fetchpriority' => $fetchpriority_val,
			]
```
Short array syntax please 😃